### ISSUE TITLE
https redirection broken with IPv6

### ISSUE THREAD

**Issue by herver** [containous/traefik#5197]
### Do you want to request a *feature* or report a *bug*?

Bug

### Did you try using a 1.7.x configuration for the version 2.0?

- [ ] Yes
- [X] No

### What did you do?
I'm evaluating IPv6 on the traefik nodes and found the following issue:
I have the following redirection set in my `http` entrypoint in order to redirect all requests to HTTPS.

```
  [entryPoints.http.redirect]
    entryPoint = "https"
```

### What did you expect to see?
I was expecting a redirect to `https://[2a02:168:xx::xx]:443/traefik/`
```
curl -v -k -6 'http://[2a02:168:xx::xx]/traefik/'
*   Trying 2a02:168:xx::xx:80...
* TCP_NODELAY set
* Connected to 2a02:168:xx::xx (2a02:168:xx::xx) port 80 (#0)
> GET /traefik/ HTTP/1.1
> Host: [2a02:168:xx::xx]> User-Agent: curl/7.65.3
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 302 Found
< Location: https://2a02:168:xx::xx]:443/traefik/
< Date: Tue, 13 Aug 2019 06:53:22 GMT
< Content-Length: 5
< Content-Type: text/plain; charset=utf-8< 
* Connection #0 to host 2a02:168:xx::xx left intact
```

### What did you see instead?
A redirect to `https://http:443://[2a02:168:xx::xx]/traefik/`

```
curl -v -k -6 'http://[2a02:168:xx::xx]/traefik/'
*   Trying 2a02:168:xx::xx:80...
* TCP_NODELAY set
* Connected to 2a02:168:xx::42 (2a02:168:xx::xx) port 80 (#0)
> GET /traefik/ HTTP/1.1
> Host: [2a02:168:xx::xx]> User-Agent: curl/7.65.3
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 302 Found
< Location: https://http:443://[2a02:168:xx::xx]/traefik/
< Date: Tue, 13 Aug 2019 06:53:22 GMT
< Content-Length: 5
< Content-Type: text/plain; charset=utf-8< 
* Connection #0 to host 2a02:168:xx::42 left intact
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.7.13
Codename:     maroilles
Go version:   go1.11.12
Built:        2019-08-08_04:46:14PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
* Running traefik in docker with discovery via the socket
* HTTP to HTTPS redirection is configured in `traefik.toml`
 
```toml
defaultEntryPoints = ["https"]

[entryPoints]
  [entryPoints.http]
    address = ":80"
  [entryPoints.http.redirect]
    entryPoint = "https"
  [entryPoints.https]
    address = ":443"
  [entryPoints.https.tls]

[docker]
...
```

**Comment by d7volker** [comment]
I can confirm the issue and tested with traefik 2.2.1

The following curl is being redirected to **https://http://[b8:a:8b:cc::a111]**

`# curl -svLk "http://[b8:a:8b:cc::a111]/"
*   Trying b8:a:8b:cc::a111:80...
* Connected to b8:a:8b:cc::a111 (b8:a:8b:cc::a111) port 80 (#0)
> GET / HTTP/1.1
> Host: [b8:a:8b:cc::a111]
> User-Agent: curl/7.70.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 301 Moved Permanently
< Location: https://http://[b8:a:8b:cc::a111]/
< Date: Sun, 24 May 2020 13:00:16 GMT
< Content-Length: 17
< Content-Type: text/plain; charset=utf-8
< 
* Ignoring the response-body
* Connection #0 to host b8:a:8b:cc::a111 left intact
* Issue another request to this URL: 'https://http//[b8:a:8b:cc::a111]/'
* Could not resolve host: http
* Closing connection 1
`

I have the following entryPoints configuration:

`entryPoints:
  web:
    address: :80
    http:
      redirections:
        entryPoint:
          to: websecure
          scheme: https
  websecure:
    address: :443
    http:
      tls:
        certResolver: letsencrypt
      middlewares:
        - secHeaders@file
`

I would expect that this request is being rejected after the redirect, because I set sniStrict to true. For ipv4 it is working as expected.

**Comment by traefiker** [comment]
Closed by #6902.

**Comment by traefiker** [comment]
Closed by #6901.
